content/blog/its-time-to-try-out-pixi/contents.lr

```diff
@@ -0,0 +1,995 @@
+jaxlib = { version = "*", build = "cuda12" }
+
+# Environments
+[tool.pixi.environments]
+cuda = ["cuda"]
+```
+
+Then, I made a `Dockerfile` that looks like the following:
+
+```Dockerfile
+# FILE: Dockerfile
+FROM ghcr.io/prefix-dev/pixi:latest
+
+WORKDIR /repo
+
+COPY pixi.lock /repo/pixi.lock
+COPY pyproject.toml /repo/pyproject.toml
+
+RUN /usr/local/bin/pixi install --manifest-path pyproject.toml --environment cuda
+
+# Entrypoint shell script ensures that any commands we run start with `pixi shell`,
+# which in turn ensures that we have the environment activated
+# when running any commands.
+COPY entrypoint.sh /repo/entrypoint.sh
+RUN chmod 700 /repo/entrypoint.sh
+ENTRYPOINT [ "/repo/entrypoint.sh" ]
+```
+
+Notice how there is an official `pixi` Docker container!
+Most crucially, it _does not_ ship with anything CUDA-related;
+it is now possible to use non-NVIDIA docker containers to run GPU code
+thanks to the many efforts of `conda-forge` community members who work for NVIDIA,
+as we can specify whether we need GPU-related packages or not
+entirely within our `conda`/`pixi` environments instead.
+
+To start, I built the docker container locally on my home lab machine:
+
+```shell
+docker build -f Dockerfile . -t pixi-cuda
+```
+
+On my home lab machine, Docker build took about 3 minutes to complete.
+The longest build step was the Pixi install command in the Dockerfile;
+the second longest build step was exporting the layers.
+
+```shell
+[+] Building 194.3s (12/12) FINISHED                      docker:defaultl
+ => [internal] load build definition from Dockerfile                0.0s
+ => => transferring dockerfile: 533B                                0.0sj
+ => [internal] load metadata for ghcr.io/prefix-dev/pixi:latest     0.4s
+ => [internal] load .dockerignore                                   0.0so
+ => => transferring context: 2B                                     0.0s
+ => [1/7] FROM ghcr.io/prefix-dev/pixi:latest@sha256:45d86bb788aaa  0.0s
+ => [internal] load build context                                   0.0s
+ => => transferring context: 118.83kB                               0.0s
+ => CACHED [2/7] WORKDIR /repo                                      0.0s
+ => [3/7] COPY pixi.lock /repo/pixi.lock                            0.1s
+ => [4/7] COPY pyproject.toml /repo/pyproject.toml                  0.1s
+ => [5/7] RUN /usr/local/bin/pixi install --manifest-path pyproj  170.9s
+ => [6/7] COPY entrypoint.sh /repo/entrypoint.sh                    0.1s
+ => [7/7] RUN chmod 700 /repo/entrypoint.sh                         0.3s
+ => exporting to image                                             22.2s
+ => => exporting layers                                            22.2s
+ => => writing image sha256:6f13e2a7b362c7b3736d4405f7f5566775320d  0.0s
+```
+
+To verify that I could indeed run CUDA-accelerated JAX, I entered into the container:
+
+```shell
+❯ docker run --gpus all -it docker.io/library/pixi-cuda /bin/bash
```

Comment:
I didn't know this existed, but after following the following guide it did in deed work :100:  https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html